services/checks/checks.go

```diff
@@ -419,6 +420,124 @@ func (s *Service) EnableChecks(checkNames []string) error {
 	return nil
 }
 
+type ChangeIntervalParams struct {
+	Name     string
+	Interval managementpb.SecurityCheckInterval
+}
+
+// ChangeInterval changes a check's interval to the value received from the UI.
+func (s *Service) ChangeInterval(params []ChangeIntervalParams) error {
+	checks := s.GetAllChecks()
+	if len(checks) == 0 {
+		return errors.New("no checks loaded")
+	}
+
+	checkMap := make(map[string]check.Check, len(checks))
+	for _, ch := range checks {
+		checkMap[ch.Name] = ch
+	}
+
+	for _, p := range params {
+		c, ok := checkMap[p.Name]
+		if !ok {
+			return errors.Errorf("check: %s not found", p.Name)
+		}
+
+		switch p.Interval {
+		case managementpb.SecurityCheckInterval_STANDARD:
+			c.Interval = check.Standard
+		case managementpb.SecurityCheckInterval_FREQUENT:
+			c.Interval = check.Frequent
+		case managementpb.SecurityCheckInterval_RARE:
+			c.Interval = check.Rare
+		case managementpb.SecurityCheckInterval_SECURITY_CHECK_INTERVAL_INVALID:
+			return errors.New("invalid security check interval")
+		default:
+			return errors.New("unknown security check interval")
+		}
+
+		s.updateCheck(c)
+
+		// since we re-run checks at regular intervals using a call
+		// to s.StartChecks which in turn calls s.collectChecks
+		// to load/download checks, we must persist any changes
+		// to check intervals in the DB so that they can be re-applied
+		// once the checks have been re-loaded on restarts.
+		cs, err := models.FindCheckStateByName(s.db.Querier, c.Name)
+		// record interval change for the first time
+		if err == reform.ErrNoRows {
+			cs, err := models.CreateCheckState(s.db.Querier, c.Name, models.Interval(c.Interval))
+			if err != nil {
+				return err
+			}
+			s.l.Debugf("Saved interval change for check: %s in DB", cs.Name)
+			break
+		}
+
+		// update interval change if already present
+		if cs != nil {
+			cs, err := models.ChangeCheckState(s.db.Querier, c.Name, models.Interval(c.Interval))
+			if err != nil {
+				return err
+			}
+			s.l.Debugf("Updated interval change for check: %s in DB", cs.Name)
+			break
+		}
+		return err
+	}
+
+	return nil
+}
+
+// updateCheck updates a check with an updated interval in the appropriate check slice.
+func (s *Service) updateCheck(newCheck check.Check) {
+	switch newCheck.Type {
+	case check.MySQLSelect:
+		fallthrough
```

Comment:
interesting case about `fallthrough` platform team uses it, but Core and Server integration teams uses comma.
https://github.com/percona/pmm-managed/blob/1b79357f4ef7037243e532c07d3cb8f6ceee7ee4/services/victoriametrics/prometheus.go#L146